src/main/kotlin/org/rust/ide/typing/assist/FunctionOrStructFixer.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.typing.assist
+
+import com.intellij.lang.SmartEnterProcessorWithFixers
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsElementTypes.IDENTIFIER
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.lang.core.psi.ext.body
+import org.rust.lang.core.psi.ext.elementType
+
+class FunctionOrStructFixer : SmartEnterProcessorWithFixers.Fixer<RsSmartEnterProcessor>() {
+
+    override fun apply(editor: Editor, processor: RsSmartEnterProcessor, element: PsiElement) {
+        if (element.elementType != IDENTIFIER) return
+
+        val parent = element.parent
+
+        val content = when {
+            parent is RsFunction && parent.valueParameterList == null && parent.body == null ->
```

Comment:
Why it shouldn't work in case of `parent.valueParameterList != null`? For example, I would expect something like this:

```rust
// Before
fn foo(a: i32, b: i32/*caret*/)

//////////////////////

// After
fn foo(a: i32, b: i32) {
    /*caret*/
}
```